### PR TITLE
Task #205 feat : File download functionality from backend

### PR DESCRIPTION
**Pull request by rajnish-dargan**

### CODE REVIEW

File: administrator/controller.php [hunk 1]
```diff
@@ -38,4 +41,88 @@ public function display($cachable = false, $urlparams = false)
...
+					$app->enqueueMessage(Text::_('COM_TJFIELDS_FILE_NOT_FOUND'), 'error');
+					$app->redirect($this->returnURL);
+				}
+
+				return;
+			}
+			else
+			{
+				$app->enqueueMessage(Text::_('JERROR_ALERTNOAUTHOR'), 'error');
+				$app->redirect($this->returnURL);
```
Comment (ankush-maherwal) [inline administrator/controller.php]: @rajnish-dargan where is `$this->returnURL` variable defined?

Comment (rajnish-dargan) [reply administrator/controller.php]: added $this->returnURL

File: administrator/controller.php [hunk 2]
```diff
@@ -31,11 +55,95 @@ public function display($cachable = false, $urlparams = false)
...
 	}
+
+	/**
+	 * Fuction to get download media file
+	 *
+	 * @return object
+	 */
+	public function getMediaFile()
+	{
+		JLoader::import("/techjoomla/media/storage/local", JPATH_LIBRARIES);
```
Comment (ankush-maherwal) [inline administrator/controller.php]: @rajnish-dargan 
need CSRF token check in each function of the controller

File: administrator/helpers/tjfields.php [hunk 3]
```diff
@@ -520,58 +526,38 @@ public static function getLanguageConstant()
...
+		Table::addIncludePath(JPATH_ADMINISTRATOR . '/components/com_tjfields/tables');
+		$fieldValueTable = Table::getInstance('Fieldsvalue', 'TjfieldsTable');
+		$fieldValueTable->load(array('id' => $data['valueId']));
 
 		$subData = new stdClass;
 		$fieldId = 0;
 
 		if ($data['isSubformField'] == 1)
 		{
-			$subData = json_decode($fieldsValueTable->value);
```
Comment (ankush-maherwal) [inline administrator/helpers/tjfields.php]: @rajnish-dargan Why this code is removed?

File: administrator/helpers/tjfields.php [hunk 4]
```diff
@@ -586,68 +572,31 @@ public function deleteFile($data)
 				if ($type[0] === 'image')
 				{
 					$deleteData = array();
-					$deleteData[] = JPATH_ROOT . $data['storagePath'] . '/' . $type[0] . '/' . $data['filePath'];
```
Comment (ankush-maherwal) [inline administrator/helpers/tjfields.php]: @rajnish-dargan we need this for UCM.
Why this is updated?

File: administrator/controller.php (hunk 2 above)
Comment (rajnish-dargan) [reply administrator/controller.php]: added CSRF token check

File: administrator/helpers/tjfields.php (hunk 3 above)
Comment (rajnish-dargan) [reply administrator/helpers/tjfields.php]: made changed as per deleteFile method of frontend

File: administrator/helpers/tjfields.php (hunk 4 above)
Comment (rajnish-dargan) [reply administrator/helpers/tjfields.php]: made changed as per deleteFile method of frontend

**Pull request merged**